daffodil-cli/src/main/scala/org/apache/daffodil/Main.scala

```diff
@@ -669,7 +674,8 @@ object Main {
       case InfosetType.JSON => Left(new JsonInfosetOutputter(os, pretty = true))
       case InfosetType.JDOM => Left(new JDOMInfosetOutputter())
       case InfosetType.W3CDOM => Left(new W3CDOMInfosetOutputter())
-      case InfosetType.SAX => Right(new DaffodilParseOutputStreamContentHandler(os, pretty = true))
+      case InfosetType.SAX => Right(new DaffodilParseOutputStreamXMLContentHandler(os, pretty = true))
+      case InfosetType.EXI => Right(new DaffodilParseOutputStreamEXIContentHandler(os))
```

Comment:
Per comments below, right here (or in a private method of Main called from here) is where the code you have in the DaffodilParseOutputStreamEXIContentHandler constructor would go. That would allow you to remove the dependency on EXI from the runtime library and put it only on the CLI library.